library/src/visualize/polygon.rs

```diff
@@ -0,0 +1,62 @@
+use crate::prelude::*;
+
+/// A closed-path polygon.
+///
+/// ## Example
+/// ```example
+/// #polygon(fill: blue, (0pt, 0pt), (10pt, 0pt), (10pt, 10pt))
+/// ```
+///
+/// Display: Polygon
+/// Category: visualize
+#[element(Layout)]
+pub struct PolygonElem {
+    /// How to fill the polygon. See the
+    /// [rectangle's documentation]($func/rect.fill) for more details.
+    pub fill: Option<Paint>,
+
+    /// How to stroke the polygon. See the [lines's
+    /// documentation]($func/line.stroke) for more details.
+    #[resolve]
+    #[fold]
+    pub stroke: Option<PartialStroke>,
+
+    /// The vertices of the polygon. The polygon automatically closes itself.
+    #[variadic]
+    pub vertices: Vec<Axes<Rel<Length>>>,
+}
+
+impl Layout for PolygonElem {
+    fn layout(
+        &self,
+        _: &mut Vt,
+        styles: StyleChain,
+        regions: Regions,
+    ) -> SourceResult<Fragment> {
+        // If there are no points in a polygon, we dont need to layout it.
```

Comment:
I would remove this and just put an `if` around the part where you construct the path:
```rust
let shape = polygon(points, self.fill(styles), stroke);
frame.push(Point::zero(), FrameItem::Shape(shape, self.span()));
```
Calling `self.vertices()` twice is probably more expensive than the whole remainder of the function because it deserializes from a Typst array.